### PR TITLE
Create `resources` table and API

### LINKED ISSUE #28: New table: resources
- table to store any resources such as documents and websites (only ever stores external URLs, no option to upload actual files)

##### Permissions by role

- public: none
- registered, no role: none
- analyst: R (where draft = false, ADDED 19/10/)
- manager: CRUD
- admin: CRUD

### PR DESCRIPTION
**Pull request by parndt**
Fixes #28 

- table to store any resources such as documents and websites (only ever stores external URLs, no option to upload actual files)

##### Permissions by role

- public: none
- registered, no role: none
- analyst: R (where draft = false, ADDED 19/10/)
- manager: CRUD
- admin: CRUD

### CODE REVIEW

**Comment by parndt** [comment]
@tmfrnz ready

File: app/models/resource.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class Resource < VersionedRecord
+  belongs_to :type, class_name: "Resourcetype"
+
+  validates :title, presence: true
+end
```
Comment (tmfrnz) [inline app/models/resource.rb:5]: will add
* `validates :resourcetype_id, presence: true` (will rename field)

also we should likely add the following in https://github.com/dumparkltd/marine-defrag-server/tree/measure-resources (not sure if needed though
* `has_many :measures, through: :measure_resources, inverse_of: :resources`

File: app/policies/resource_policy.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  def permitted_attributes
+    [
+      :access_date,
+      :description,
+      :draft,
+      :private,
+      :publication_date,
+      :status,
+      :title,
+      :type_id,
```
Comment (tmfrnz) [inline app/policies/resource_policy.rb:11]: will change to resourcetype_id for consistency

File: app/serializers/resource_serializer.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  include FastVersionedSerializer
+
+  attributes :access_date,
+    :description,
+    :draft,
+    :private,
+    :publication_date,
+    :status,
+    :title,
+    :type_id,
```
Comment (tmfrnz) [inline app/serializers/resource_serializer.rb:11]: again: will change to resourcetype_id for consistency

File: db/schema.rb [hunk 4]
```diff
@@ -36,9 +36,11 @@
     t.bigint "updated_by_id"
     t.datetime "created_at", precision: 6, null: false
     t.datetime "updated_at", precision: 6, null: false
+    t.bigint "source_id"
```
Comment (tmfrnz) [inline db/schema.rb:39]: will rename to `resource_id`

File: db/schema.rb [hunk 5]
```diff
@@ -351,6 +353,25 @@
     t.index ["framework_id"], name: "index_recommendations_on_framework_id"
   end
 
+  create_table "resources", force: :cascade do |t|
+    t.string "title", null: false
+    t.text "description"
+    t.text "url"
+    t.bigint "type_id", null: false
```
Comment (tmfrnz) [inline db/schema.rb:360]: will rename to `resourcetype_id`

**Review by tmfrnz — APPROVED** [review]

**Pull request merged**